### ISSUE TITLE
Ignite falling off too early

### ISSUE THREAD

**Issue by Nogarthemage** [elysium-project/server#2002]
fact: Ignite duration is currently bugged, clearly visible in these 2 videos i personaly made:
-- https://www.youtube.com/watch?v=NOxX4fhB6tA
-- https://www.youtube.com/watch?v=8FaOrPoZFJQ
- Character: Nogar - Server: Darrowshire

i cast 3 fireballs, all consecutive hit and with each hit the timer should reset to 4 seconds, wich it doesnt.

speculation: the duration of the ignite timer is basing itself off 'amount of ticks left' and not '4 seconds'

thus allowing a loss of up to 1.9 seconds on your ignite debuff resulting in a huge dps loss due to the player not being able to refresh their existing ignite in this timeframe

**Comment by Esrath** [comment]
Ignite requires crititical hits to be reapplied, it is reapplied for 4 seconds which means 3 seconds on tooltip since 0 counts as 1. You did 3 crits and 1 hit and it reapplied for those 3 crits but not on the last hit. Am I missing something?

**Comment by sickheals** [comment]
The third crit doesn't cause the debuff to go from 2 to 3 stacks. It hit before ignite was gone, but it did not reapply the debuff and did not increase the damage from the ignite.

**Comment by Nogarthemage** [comment]
i do say it resets to 4 on each crit? why do u say i dont

**Comment by sickheals** [comment]
> i cast 3 fireballs, all consecutive hit and with each hit the timer should reset to 4 seconds, wich it doesnt.

**Comment by Nogarthemage** [comment]
@Esrath the point is that 3 consecutive fireball hits did not give a 3 stack ignite, it gave a 2 stack ignite and the 3rd reset the ignite back to 0, but as you can see on the recievers PoV video the ignite doesnt actualy fall off, it just resets to 1 stack, when in reality it should stack the ignite to 3

**Comment by Sirael11** [comment]
@Nogarthemage What you write in your last comment is what happens in the video. It's just that your original description is misleading.

So it's not a problem with the debuff duration since you clearly still have the 2 stack debuff when the 3rd fireball hits. The problem has to be in the logic which checks what the new stack count after a crit is.

**Comment by Nogarthemage** [comment]
hmm, well imho if you look at an exapmle where 2 fireballs crit and the 3rd one doesnt, its very visible that the ignite isnt falling off before the 3rd hits, but when it crits it forces off the ignite and puts a new one up, thats why theres also no aura loss between the transmission of 2 --> 1 stack in the video (at 13 seconds you can see in the bottom right that ignite falls of https://puu.sh/x6Ema/dad0631ee2.png)
but with the 2 -- > stack and the 3rd crit this aura doesnt fall off and re-applies. its just resets to 1 instead of stacking.

**Comment by sickheals** [comment]
Yes, you're repeating what @Sirael11 and I have already said. Thanks.

**Comment by Nogarthemage** [comment]
what?

**Comment by Nogarthemage** [comment]
you're the one not reading my initial post correct because ive had the same stance all this time.

**Comment by sickheals** [comment]
So to summarize:
1. The third crit doesn't cause the debuff to go from 2 to 3 stacks. It hit before ignite was gone, but it did not reapply the debuff and did not increase the damage from the igntite.
2. You were wrong about the timer resetting. It does reset after each crit.

**Comment by GuybrushGit** [comment]
Nothing in your video is bugged.

Crits do not reset the buff duration to 4 secs, if they did ignite would never proc in a raid since the timer would constantly be resetting.
They reset the number of ticks until it expires, so if you crit 0.1 seconds before a tick the ignite will expire in 2.1 seconds. This is the proper ignite behaviour.

**Comment by sickheals** [comment]
Perhaps I'm not understanding correctly, but the ignite should stack and should increase the value of the consecutive ignites: http://wowwiki.wikia.com/wiki/Ignite_(old)

**Comment by sickheals** [comment]
I also found this page that describes how old ignite worked: https://criticalqq.wordpress.com/2008/09/23/ignite-the-new-blink/

**Comment by GuybrushGit** [comment]
Those sources describe how ignite is currently implemented.

**Comment by Nogarthemage** [comment]
i get what u say Guy, and you are correct, what i said of them resetting to 4 indeed isnt vanilla-like, but i have still noticed while playing that the enemy clearly has stacks and my current crit pushes off the and resets the ignite currectly on the enemy, my new theory is that ignite ticking is not affected by the lag a player has and thus ticks a bit faster than the player can generater ticks, resulting in the 3-consecutive fireballs not stacking to 3. my second guess is that due to the tick system blizzard used up untill wod (i think its wod) where spells are sent in packets of 400 ms, that there the crit and ignite come in at the same time and thus making it able to stack to 3 instead of resetting
